internal/indexer/definitions/sugoimusic.yaml

```diff
@@ -0,0 +1,87 @@
+#id: sugoimusic
+name: SugoiMusic
+identifier: sugoimusic
+description: SugoiMusic (SM) is a private torrent tracker for MUSIC
+language: en-us
+urls:
+  - https://sugoimusic.me/
+privacy: private
+protocol: torrent
+supports:
+  - irc
+source: gazelle
+settings:
+  - name: authkey
+    type: secret
+    required: true
+    label: Auth key
+    help: Right click DL on a torrent and get the authkey.
+
+  - name: torrent_pass
+    type: secret
+    required: true
+    label: Torrent pass
+    help: Right click DL on a torrent and get the torrent_pass.
+
+irc:
+  network: SugoiMusic
+  server: irc.sugoimusic.me
+  port: 6697
+  tls: true
+  channels:
+    - "#sugoi-announce"
+  announcers:
+    - Yuki
+  settings:
+    - name: nick
+      type: text
+      required: true
+      label: Nick
+      help: Bot nick. Eg. user-autodl
+
+    - name: auth.account
+      type: text
+      required: false
+      label: NickServ Account
+      help: NickServ account. Make sure to group your user and bot.
+
+    - name: auth.password
+      type: secret
+      required: false
+      label: NickServ Password
+      help: NickServ password
+
+    - name: invite_command
+      type: secret
+      default: "Yuki enter #sugoi-announce USERNAME IRCKEY"
+      required: true
+      label: Invite command
+      help: Invite auth with Yuki. Replace USERNAME and IRCKEY.
+
+  parse:
+    type: multi
+    lines:
+      - test:
+          - "[Album] Artist – Album Name [FLAC / 2017 / catnumber] - https://sugoimusic.me/torrents.php?id=000001 / https://sugoimusic.me/torrents.php?action=download&id=000011 - japanese,pop,female.vocalist Uploaded by: Uploader"
+          - "[Single] Artist – Album Name [MP3] - https://sugoimusic.me/torrents.php?id=000002 / https://sugoimusic.me/torrents.php?action=download&id=000021 - japanese,pop,dance Uploaded by: Uploader"
+          - "[TV Music] TV Show – Title [TS / MPEG-2 / 1920x1080 / AAC / Japanese / NoSubs] - https://sugoimusic.me/torrents.php?id=000003 / https://sugoimusic.me/torrents.php?action=download&id=000031 - japanese,live Uploaded by: Uploader"
+        pattern: '(\[(.*)\] .*? – .+? \[((?:.*?)?(?: \/ )?(\d{4})?(?: \/ )?\w+)\]) - (https?://.+/).+id=(\d+) \/ https?://.+action=download&id=(\d+) - (.*) Uploaded by: (.*)'
+        vars:
+          - torrentName
+          - category
+          - releaseTags
+          - year
+          - baseUrl
+          - groupId
+          - torrentId
+          - tags
+          - uploader
+      - test:
+          - "FREELEECH - https://sugoimusic.me/torrents.php?id=000001 / https://sugoimusic.me/torrents.php?action=download&id=000011"
+        pattern: '(FREELEECH)? - .*'
```

Comment:
I'm not sure if this is a proper pattern for an optional second line showing a torrent is freeleech, but it seems to be working.

Reply:
Looks correct based on how we do it here: https://github.com/autobrr/autobrr/blob/feat(feeds)-add-force-run/internal/indexer/definitions/alpharatio.yaml#L51-L73

Reply:
@Kuredant Is that line totally optional? Not sure if we support that 🤔 

Have you tested it? If so, anything weird in the logs? Does it match like it should?

Reply:
@zze0s The second line appears to be optional. It's only there when a torrent is freeleech.

> Have you tested it? If so, anything weird in the logs? Does it match like it should?

It seems to be working. With the following IRC lines:

```
[TV Music] TV Show – Title [TS / MPEG-2 / 1440x1080 / AAC / Japanese / NoSubs] - https://sugoimusic.me/torrents.php?id=000010 / https://sugoimusic.me/torrents.php?action=download&id=000011 - japanese,live Uploaded by: Uploader
FREELEECH - https://sugoimusic.me/torrents.php?id=000010 / https://sugoimusic.me/torrents.php?action=download&id=000011
[Single] Artist – Album Name [FLAC] - https://sugoimusic.me/torrents.php?id=000020 / https://sugoimusic.me/torrents.php?action=download&id=000021 - japanese,pop,female.vocalist Uploaded by: Uploader
```

I'm getting the following trace output from one of my filters:

```
{"level":"trace","module":"filter","time":"2023-11-13T13:21:05Z","message":"filter.Service.CheckFilter: checking filter: cross-seed for release: &{ID:0 FilterStatus:PENDING Rejections:[] Indexer:sugoimusic FilterName:cross-seed Protocol:torrent Implementation:IRC Timestamp:2023-11-13 13:21:05.490159548 +0000 UTC m=+3.895157255 InfoURL:https://sugoimusic.me/torrents.php?id=000010&torrentid=000011 DownloadURL:https://sugoimusic.me/torrents.php?action=download&id=000011&authkey=XXX&torrent_pass=XXX MagnetURI: GroupID: TorrentID:200421 TorrentTmpFile: TorrentDataRawBytes:[] TorrentHash: TorrentName:[TV Music] TV Show – Title [TS / MPEG-2 / 1440x1080 / AAC / Japanese / NoSubs] Size:0 Title:TV Music Description: Category:TV Music Categories:[] Season:0 Episode:0 Year:0 Resolution: Source:SP Codec:[MPEG-2] Container:mpg HDR:[] Audio:[AAC] AudioChannels: Group:NoSubs Region: Language:[JAPANESE] Proper:false Repack:false Website: Artists: Type: LogScore:0 Origin: Tags:[japanese live] ReleaseTags:TS / MPEG-2 / 1440x1080 / AAC / Japanese / NoSubs Freeleech:true FreeleechPercent:100 Bonus:[Freeleech] Uploader:Uploader PreTime: Other:[] RawCookie: AdditionalSizeCheckRequired:false FilterID:12 Filter:0xc001889200 ActionStatus:[]}"}
{"level":"trace","module":"filter","time":"2023-11-14T23:43:18Z","message":"filter.Service.CheckFilter: checking filter: cross-seed for release: &{ID:0 FilterStatus:PENDING Rejections:[] Indexer:sugoimusic FilterName:cross-seed Protocol:torrent Implementation:IRC Timestamp:2023-11-14 23:43:18.35319562 +0000 UTC m=+123736.758193335 InfoURL:https://sugoimusic.me/torrents.php?id=000020&torrentid=000021 DownloadURL:https://sugoimusic.me/torrents.php?action=download&id=000021&authkey=XXX&torrent_pass=XXX MagnetURI: GroupID: TorrentID:200426 TorrentTmpFile: TorrentDataRawBytes:[] TorrentHash: TorrentName:[Album] Artist – Album Name [FLAC] Size:0 Title:Kay Ishiguro – Amant Description: Category:Album Categories:[] Season:0 Episode:0 Year:0 Resolution: Source: Codec:[] Container: HDR:[] Audio:[FLAC] AudioChannels: Group: Region: Language:[] Proper:false Repack:false Website: Artists: Type: LogScore:0 Origin: Tags:[japanese pop female.vocalist] ReleaseTags:FLAC Freeleech:false FreeleechPercent:0 Bonus:[] Uploader:Uploader PreTime: Other:[] RawCookie: AdditionalSizeCheckRequired:false FilterID:12 Filter:0xc001373200 ActionStatus:[]}"}
```

The `Freeleech` and `FreeleechPercent` are properly set to `true` and `100` for the first release and to `false` and `0` for the second.

Reply:
Okay, that sounds promising! I think we can probably merge this then if it all seem to work on your end 👍

Reply:
@zze0s Actually, after some more testing, this seems to not be working as expected for non-freeleech releases: if a release has only one line, it will expect two lines and re-grab the previous release instead.

I'll just remove multiline matching and ignore the freeleech flag for the time being, until we get proper support for optional lines.

Reply:
As I suspected! Will have to think about how we can add optional lines.